src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> previous;
+        private Node<T> next;
+
+        private Node(T value, Node<T> previous, Node<T> next) {
+            this.value = value;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
+    private int size;
+    private Node<T> headNode;
+    private Node<T> tailNode;
+
+    private Node<T> getNodeFromIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("wrong index"
+                    + "\n"
+                    + "Mast be from 0 to " + (size() - 1));
+        }
+        Node<T> searchNode = headNode;
+
+        if (index < (size >> 1)) {
+            for (int i = 0; i < index; i++) {
+                searchNode = searchNode.next;
+            }
+        } else {
+            searchNode = tailNode;
+            for (int i = size() - 1; i > index; i--) {
+                searchNode = searchNode.previous;
+            }
+        }
+        return searchNode;
+
+    }
+
+    private void addToEmpty(T value) {
+        Node<T> newNode = new Node<>(value, null, null);
+        headNode = newNode;
+        tailNode = newNode;
+        size++;
+    }
+
+    private void addFirst(T value) {
+        if (size == 0) {
```

Comment:
do we need this `if` here:
if (size == 0) { 
addToEmpty(value);
        }

Reply:
No, but it like insurance

Reply:
Without checking i get NullPointerException

Reply:
Teste failed